woocommerce/wc-admin
# Add Typescript and page objects to E2E suite

Partly addresses #6313 

This is basically the original https://github.com/woocommerce/woocommerce-admin/pull/6484 PR just without playwright.
The biggest changes are using TypeScript and Page objects.

### Screenshots

<img width="1027" alt="Screen Shot 2021-03-12 at 5 04 09 PM" src="https://user-images.githubusercontent.com/2240960/110998290-f96b9180-8354-11eb-812b-71bf66bbd04d.png">

### Detailed test instructions:

- Load this PR and run `npm install`
- Run `npm run docker:up`
- Run `npm run test:e2e`, it should run all the tests and pass
- Run `npm run test:e2e-dev`, it should show a chromium browser and run the tests (one of the tests might fail the second time around, given we can't reset the DB yet).
- Run `npm run test:e2e tests/e2e/specs/activate-and-setup/setup-wizard.test.ts`, it should just run that test file.
